docs/opup.rst

```diff
@@ -0,0 +1,85 @@
+.. _opup:
+
+Budget Increase Utility
+========================
+
+Some opcode budget is consumed during execution of every Algorand Smart Contract because every teal
+instruction has a corresponding cost. In order for the evaluation to succeed, the budget consumed must not
+exceed the budget provided. This constraint may introduce a problem for expensive contracts that quickly
+consume the initial budget of 700. The OpUp budget increase utility provides a workaround using inner
+transactions. Note that there is a context specific limit to the number of inner transactions issued in a
+transaction group so budget cannot be increased arbitrarily. The available budget when using the OpUp
+utility will need to be high enough to execute the teal code that issues the inner transactions. A budget
+of ~20 is enough for most use cases.
+
+Usage
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The :any:`pyteal.OpUp` utility is available in two modes: :any:`Explicit` and :any:`OnCall`:
+
+================= ================================================================================
+OpUp Mode         Description
+================= ================================================================================
+:code:`Explicit`  Calls a user provided external app when more budget is requested.
+:code:`OnCall`    Creates the app to be called when more budget is requested.
+================= ================================================================================
+
+
+:any:`Explicit` has the benefit of constructing more lightweight inner transactions but requires the
+target app ID to be provided in the foreign apps array field of the transaction and the :any:`pyteal.OpUp`
+constructor in order for it to be accessible. :any:`OnCall` is easier to use but has slightly more overhead
+because the target app must be created and deleted during the evaluation of an app call.
```

Comment:
@algoidurovic Minor:  _Suggests_ adding commas for readability.

```suggestion
:any:`Explicit` has the benefit of constructing more lightweight inner transactions, but requires the
target app ID to be provided in the foreign apps array field of the transaction and the :any:`pyteal.OpUp`
constructor in order for it to be accessible. :any:`OnCall` is easier to use, but has slightly more overhead
because the target app must be created and deleted during the evaluation of an app call.
```